src/components/SegmentedControl/index.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { ReactNode } from 'react';
+import PropTypes from 'prop-types';
+import RenderIf from '../RenderIf';
+import { BaseProps } from '../types';
+import {
+    StyledSegmentedControlContainer,
+    StyledSegmentedControlLabel,
+    StyledSegmentedControlOptionsWrapper,
+} from './styled';
+import SegmentedControlOption from './SegmentedControlOption';
+
+interface Option {
+    label: ReactNode;
+    value: string;
+    icon?: ReactNode;
+}
+
+interface Props extends BaseProps {
+    label?: string;
```

Comment:
The label prop should be a ReactNode, we need to support this for things like internationalization